### PR TITLE
land: Build kernel from source

### PR DESCRIPTION
**Pull request by yangyangnau**
If the kernel source is exist, build kernel from source.
Fix board/kernel.mk.

### CODE REVIEW

File: board/kernel.mk [hunk 1]
```diff
@@ -5,6 +5,11 @@ BOARD_KERNEL_CMDLINE := console=ttyHSL0,115200,n8 androidboot.console=ttyHSL0 an
 BOARD_KERNEL_CMDLINE += androidboot.selinux=permissive
 BOARD_KERNEL_PAGESIZE :=  2048
 BOARD_MKBOOTIMG_ARGS := --ramdisk_offset 0x01000000 --tags_offset 0x00000100
-TARGET_KERNEL_CONFIG := cyanogenmod_land_defconfig
+BOARD_KERNEL_IMAGE_NAME := Image.gz-dtb
+TARGET_KERNEL_APPEND_DTB := true
+TARGET_KERNEL_ARCH := arm64
 TARGET_KERNEL_SOURCE := kernel/xiaomi/msm8937
+TARGET_KERNEL_CONFIG := cyanogenmod_land_defconfig
+TARGET_KERNEL_CROSS_COMPILE_PREFIX := aarch64-linux-android-
```
Comment (zwliew) [inline board/kernel.mk]: no-op

File: board/kernel.mk [hunk 2]
```diff
@@ -5,6 +5,11 @@ BOARD_KERNEL_CMDLINE := console=ttyHSL0,115200,n8 androidboot.console=ttyHSL0 an
 BOARD_KERNEL_CMDLINE += androidboot.selinux=permissive
 BOARD_KERNEL_PAGESIZE :=  2048
 BOARD_MKBOOTIMG_ARGS := --ramdisk_offset 0x01000000 --tags_offset 0x00000100
-TARGET_KERNEL_CONFIG := cyanogenmod_land_defconfig
+BOARD_KERNEL_IMAGE_NAME := Image.gz-dtb
+TARGET_KERNEL_APPEND_DTB := true
+TARGET_KERNEL_ARCH := arm64
```
Comment (zwliew) [inline board/kernel.mk]: no-op

File: board/kernel.mk [hunk 3]
```diff
@@ -5,6 +5,11 @@ BOARD_KERNEL_CMDLINE := console=ttyHSL0,115200,n8 androidboot.console=ttyHSL0 an
...
 TARGET_KERNEL_SOURCE := kernel/xiaomi/msm8937
+TARGET_KERNEL_CONFIG := cyanogenmod_land_defconfig
```
Comment (zwliew) [inline board/kernel.mk]: no point moving this line

File: board/kernel.mk [hunk 4]
```diff
@@ -5,6 +5,11 @@ BOARD_KERNEL_CMDLINE := console=ttyHSL0,115200,n8 androidboot.console=ttyHSL0 an
 BOARD_KERNEL_CMDLINE += androidboot.selinux=permissive
 BOARD_KERNEL_PAGESIZE :=  2048
 BOARD_MKBOOTIMG_ARGS := --ramdisk_offset 0x01000000 --tags_offset 0x00000100
-TARGET_KERNEL_CONFIG := cyanogenmod_land_defconfig
+BOARD_KERNEL_IMAGE_NAME := Image.gz-dtb
+TARGET_KERNEL_APPEND_DTB := true
```
Comment (zwliew) [inline board/kernel.mk]: this seems like a no-op as long as BOARD_KERNEL_IMAGE_NAME := Image.gz-dtb is set

**Comment by zwliew** [comment]
Make sure to get rid of the custom boot image makefile

**Comment by zwliew** [comment]
You can use this for your cherry-picks, but I don't really intend on merging this. We are yet to have a proper kernel to use.

**Comment by yangyangnau** [comment]
Ok. I see!

Update: Using prebuilt kernel as a switch
If the prebuilt kernel is exist, use it.
If remove the prebuild kernel, build kernel from source.

It's necessary keeping TARGET_KERNEL_CROSS_COMPILE_PREFIX.
Keep TARGET_KERNEL_ARCH too for good looking.

File: board/kernel.mk (hunk 4 above)
Comment (yangyangnau) [inline board/kernel.mk]: Sure.

File: board/kernel.mk (hunk 3 above)
Comment (yangyangnau) [inline board/kernel.mk]: ok

File: board/kernel.mk (hunk 1 above)
Comment (yangyangnau) [inline board/kernel.mk]: It's necessary keeping TARGET_KERNEL_CROSS_COMPILE_PREFIX.
Or will pass wrong "CROSS_COMPILE=arm-eabi-" to kernel make command.

File: board/kernel.mk (hunk 2 above)
Comment (yangyangnau) [inline board/kernel.mk]: Sure.

**Pull request closed without merging**